scripts/local-universe.py

```diff
@@ -208,43 +188,100 @@ def include_revision(package_name, revision_path):
 
             # Include only the first acceptable revision
             for revision_path in revision_paths:
-                if include_revision(package_path.name, revision_path):
-                    yield (package_path.name, revision_path)
-                    break
+                package_json = load_json(revision_path / 'package.json')
+                if include_revision(
+                    package_json,
+                    pending_packages,
+                    only_selected,
+                    dcos_version
+                ):
+                    # *Mutation*. We enumerated the package so let's remove
+                    # it from our pending list if it exists. It may not exists
+                    # if --selected is used.
+                    key = [package_json['name'], package_json['version']]
+                    if key in pending_packages:
+                        pending_packages.remove(key)
+
+                    yield (
+                        package_json['name'],
+                        package_json['version'],
+                        revision_path
+                    )
+
+    if pending_packages:
+        print("Error: couldn't find the following packages")
+        print(pending_packages)
+        sys.exit(1)
+
+
+def include_revision(package_json, packages, only_selected, dcos_version):
+    version_pass = version_check(package_json, dcos_version)
+    selected_pass = selected_check(package_json, packages, only_selected)
+
+    return version_pass and selected_pass
+
 
+def version_check(package_json, dcos_version):
+    if dcos_version:
+        raw_version = package_json.get('minDcosReleaseVersion')
+        if raw_version:
+            min_version = distutils.version.LooseVersion(raw_version)
+            if dcos_version < min_version:
+                return False
+    return True
 
-def enumerate_http_resources(package, package_path, skip_images, skip_cli):
-    resource_path = package_path / 'resource.json'
-    with resource_path.open(encoding='utf-8') as json_file:
-        resource = json.load(json_file)
+
+def selected_check(package_json, packages, only_selected):
+    package_name = package_json['name']
+    package_version = package_json['version']
+
+    if only_selected:
+        return package_json.get('selected', False)
+
+    return [package_name, package_version] in packages
```

Comment:
Yeah. I ended up changing the semantic of an empty `packages`. We were interpreting as download everthing. I don't think that is what the user wants.